📑 Docs: Update README and issues ahead of v1.0 release

azigler opened the pull request:
Fixes #104

azigler commented:
@crspeller Right now we have these listed as the plugin's "Usages":

- Streaming Conversation
- Thread Summarization
- Answer questions about Threads
- Chat anywhere
- Create meeting summary
- Personalisation
- User lookup (OpenAI exclusive)
- Channel posts lookup (OpenAI exclusive)
- GitHub integration (OpenAI exclusive, requires GitHub plugin)
- React for me

1. Are there newer, better usages that we should highlight here for v1.0?

1. Also, are there any usages that we should remove or edit?

1. Thoughts on moving the OpenAI-exclusive things to a dedicated section?

azigler commented:
@asaadmahmood The README would benefit from a new header image. The "Usage" section might get changed or altered (see my message above) so hold off on re-doing those images for now. But once I confirm the usage list, I would also like your help with getting nice GIFs embedded here for them.

Right now, the README is hard to follow because of all the floating, dis-similar screenshots, in my opinion. Aesthetically, I'd like to clean it up a bit as I trim down the copy.

asaadmahmood commented:
Sounds good @azigler Let me know when we're ready to update the images.

azigler commented:
@crspeller @jespino I have simplified all of the documentation ahead of v1.0 and it's ready for review.

@asaadmahmood Could you propose a header image that we can embed in the README and add as a [social media preview image](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/customizing-your-repositorys-social-media-preview). Refer to the link for size recommendations.

jprusch commented:
You might want to add some prerequisites to the readme. As far as I know, PostgreSQL is needed to run this plugin or has this changed since the initial releases?

azigler commented:
Thank you for the feedback @jprusch, I have added a note that PostgreSQL is required. I originally omitted it because PostgreSQL is the official supported database for Mattermost, but I have added it here for further clarity since this plugin is PostgreSQL-specific.

jprusch commented:
It's actually the ONLY plugin that is pure PostgreSQL. I was checking the migration path to PostgreSQL to test a little more on this plugin, but it''s not working out of the box & we've got no knowledge on PostgreSQL. As I understand the path to only support one database on Mattermost server in the future, current plugins should support both platforms as long as it's supported by the Mattermost server. You won't get any feedback from customers/users which are on MySQL...

crspeller commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,22 +1,27 @@
+
+<!-- Explain why these changes are needed. -->
+
+## Types of changes
```
This feels unnecessary.

crspeller commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,22 +1,27 @@
+- [ ] Documentation update (including new documentation and translations)
+- [ ] Other (please describe):
+
+## Test plan
```
Remove.

crspeller commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,22 +1,27 @@
+
+<!-- Describe how the changes were tested. -->
+
+## Additional context
```
Why not just explain in the description?

crspeller commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,22 +1,27 @@
+
+<!-- Provide a clear and concise description of the changes made in this pull request. -->
+
+## Motivation
```
Why not just explain in the description?

crspeller commented on LICENSE:
```diff
@@ -1,201 +1,192 @@
-   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-   See the License for the specific language governing permissions and
-   limitations under the License.
+# License
```
I don't think you should modify the license like that.

crspeller commented on README.md:
```diff
@@ -1,185 +1,48 @@
-### Mattermost
-- [Mattermost Troubleshooting Discussion on Mattermost Forum](https://forum.mattermost.com/c/trouble-shoot/16)
-- [Mattermost "Peer-to-peer Help" channel on Mattermost Community server](https://community.mattermost.com/core/channels/peer-to-peer-help)
+Exploring AI-powered collaboration for your organization? Check out [OpenOps](https://openops.mattermost.com), an open source framework for experimenting with AI workflows in secure environments.
```
Lets drop the mention of OpenOps.

crspeller commented on docs/configuration-guide.md:
```diff
@@ -0,0 +1,58 @@
+
+The Mattermost AI Plugin can support any LLM provider that is OpenAI-compatible such as [LocalAI](https://github.com/go-skynet/LocalAI).
+
+1. Deploy your model on LocalAI
```
```suggestion
1. Deploy your model on LocalAI or equivalent
```

crspeller commented on docs/configuration-guide.md:
```diff
@@ -0,0 +1,58 @@
+- [Ask Sage (ChatGPT)](#ask-sage-chatgpt)
+- [OpenAI Compatible](#openai-compatible)
+
+## OpenAI (ChatGPT)
```
```suggestion
## OpenAI
```

It's specifically not ChatGPT it's though the API.

crspeller commented on docs/configuration-guide.md:
```diff
@@ -0,0 +1,58 @@
+1. In the **API Key** field, enter your Azure resource API key
+1. In the **Default Model** field, enter the model name (e.g., `gpt-35-turbo`)
+
+## Ask Sage (ChatGPT)
```
```suggestion
## Ask Sage (experimental, streaming not supported)
```
Also not ChatGPT

crspeller commented on docs/configuration-guide.md:
```diff
@@ -0,0 +1,58 @@
+
+## Ask Sage (ChatGPT)
+
+Ask Sage is currently supported as an experimental feature. Token-based security is not yet available via the Ask Sage API, and server configuration would require securing the Mattermost server configuration data store, which will contain username and password in plaintext.
```
Add a note here that streaming is also not supported.

crspeller commented on docs/developer-setup-guide.md:
```diff
@@ -0,0 +1,23 @@
+1. Access Mattermost and configure the Mattermost AI Plugin:
+
+   1. Log in to Mattermost as an administrator
+   1. Upload the Mattermost AI Plugin via **System Console** ➡️ **Plugin Management**
```
These last two steps are done by that make deploy command. It will upload and enable automatically. We can just remove this section.

crspeller commented on docs/usage.md:
```diff
@@ -0,0 +1,26 @@
+
+This document demonstrates potential AI workflows with `mattermost-plugin-ai`. Experimenting with your own AI workflow? [Submit a pull request](https://github.com/mattermost/mattermost-plugin-ai/pulls) and add it to the list! 🚀
+
+Exploring AI-powered collaboration for your organization? Check out [OpenOps](https://openops.mattermost.com), an open source framework for experimenting with AI workflows in secure environments.
```
Remove I think it's confusing.

crspeller commented on docs/usage.md:
```diff
@@ -0,0 +1,26 @@
+| **Meeting Summarization**                      | ![Meeting Summarization](https://github.com/mattermost/mattermost-plugin-ai/blob/master/img/meeting_summary.png?raw=true)                                                                                            | Create meeting summaries! Designed to work with the [Mattermost Calls plugin](https://github.com/mattermost/mattermost-plugin-calls)  ...
+| **Chat with AI Bots**                          | ![Chat with AI Bots](https://github.com/mattermost/mattermost-plugin-ai/blob/master/img/chat_anywhere.png?raw=true)                                                                                                  | End users can interact with the AI bot in any discussion thread by mentioning AI bot with an `@` prefix, as they would get the attent ...
+| **Sentiment Analysis**                         | [![React for me](https://github.com/mattermost/openops/assets/3191642/56bf132a-b834-46a3-882c-9b1f38a9f9fc)](https://github.com/mattermost/mattermost-plugin-ai/assets/3191642/5282b066-86b5-478d-ae10-57c3cb3ba038) | Use the "React for me" menu option to have the AI bot analyze the sentiment of messages use its conclusion to deliver an emoji reacti ...
+| **Reinforcement Learning from Human Feedback** | ![RLHF](https://github.com/mattermost/openops/assets/3191642/ec330f7e-2aba-4370-bf21-e585a793160e)                                                                                                                   | Bot posts are distinguished from human posts by having 👍 👎 icons available for human end users to signal whether the AI response was  ...
```
Remove, not in V1

crspeller commented on docs/usage.md:
```diff
@@ -0,0 +1,26 @@
+| **Reinforcement Learning from Human Feedback** | ![RLHF](https://github.com/mattermost/openops/assets/3191642/ec330f7e-2aba-4370-bf21-e585a793160e)                                                                                                                   | Bot posts are distinguished from human posts by having 👍 👎 icons available for human end users to signal whether the AI response was  ...
+| **Personalization** | ![Personalisation](https://github.com/mattermost/mattermost-plugin-ai/raw/master/img/personalization.png)                                                                                                                   | Context such as the current channel and user are supplied to the LLM when you make requests. Allowing customization of responses. |
+
+## OpenAI
```
All this is dependent on function calling so it should work on Azure as well.

crspeller reviewed (changes requested):
Lots of good updates! Have a few comments.

crspeller commented:
> Are there newer, better usages that we should highlight here for v1.0?

Yes we should update the screenshots and add stuff like the RHS. 

> Thoughts on moving the OpenAI-exclusive things to a dedicated section?

Not sure I like that, I think they can be together as long as the limitations are clear.

azigler commented:
> It's actually the ONLY plugin that is pure PostgreSQL. I was checking the migration path to PostgreSQL to test a little more on this plugin, but it''s not working out of the box & we've got no knowledge on PostgreSQL. As I understand the path to only support one database on Mattermost server in the future, current plugins should support both platforms as long as it's supported by the Mattermost server. You won't get any feedback from customers/users which are on MySQL...

Hey @jprusch thank you for this additional context. We updated the [MySQL ➡️ PostgreSQL migration guide](https://docs.mattermost.com/deploy/postgres-migration.html) in the last month in https://github.com/mattermost/docs/pull/6820 and I was wondering if you've had a chance to review that updated version. We recently helped several customers successfully migrate from MySQL 8 and updated the guide with their feedback. If you're encountering issues outside of the guide, we'd be more than happy to look into it with you.
